packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,367 @@
+import { getHintType } from '~components/Input/BaseInput/BaseInput';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const _FileUpload: React.ForwardRefRenderFunction<BladeElementRef, FileUploadProps> = (
+  {
+    name,
+    accept,
+    uploadType = 'single',
+    onChange,
+    onPreview,
+    onRemove,
+    onCancel,
+    onDrop,
+    isDisabled,
+    isRequired,
+    necessityIndicator,
+    fileList,
+    testID,
+    label,
+    labelPosition = 'top',
+    accessibilityLabel,
+    validationState,
+    helpText,
+    errorText,
+    maxCount,
+    maxSize,
+    ...styledProps
+  },
+  ref,
+): React.ReactElement => {
+  const { platform } = useTheme();
+  const [selectedFiles, setSelectedFiles] = useState<BladeFileList>(fileList ?? []);
+  const [errorMessage, setErrorMessage] = useState(errorText);
+  const [internalValidationState, setInternalValidationState] = useState('none');
+  const [isActive, setIsActive] = useState(false);
+
+  const isMultiple = uploadType === 'multiple';
+  const isOneFileSelectedWithSingleUpload = !isMultiple && selectedFiles.length === 1;
+  const inputLabelPosition = platform === 'onMobile' ? 'top' : labelPosition;
+  const isLabelLeftPositioned = inputLabelPosition === 'left';
+  const willRenderHintText = isOneFileSelectedWithSingleUpload
+    ? false
+    : Boolean(helpText) || Boolean(errorMessage);
+
+  const showError = validationState === 'error' || internalValidationState === 'error';
+  const showHelpText = !showError && helpText;
+  const accessibilityText =
+    accessibilityLabel ?? `,${showError ? errorText : ''} ${showHelpText ? helpText : ''}`;
+  const { inputId, labelId, helpTextId, errorTextId } = useFormId('fileuploadinput');
+
+  const accessibilityProps = makeAccessible({
+    required: Boolean(isRequired),
+    invalid: Boolean(showError),
+    disabled: Boolean(isDisabled),
+    describedBy: helpTextId,
+  });
+
+  // In control mode attach a unique id to each file if not provided
+  useMemo(() => {
+    for (const file of selectedFiles) {
+      if (!file.id) {
+        file.id = `${new Date().getTime().toString()}${Math.floor(Math.random() * 1000000)}`;
+      }
+    }
+  }, [selectedFiles]);
+
+  const handleFilesChange = useCallback((inputFiles: BladeFileList) => {
+    setSelectedFiles((prevFiles) => {
+      if (prevFiles.length > 0) {
+        const allFiles = [...prevFiles, ...inputFiles];
+        return allFiles;
+      }
+
+      return inputFiles;
+    });
+  }, []);
+
+  const validateFiles = (inputFiles: BladeFileList, allFiles: BladeFileList): boolean => {
+    if (accept && inputFiles.some((file) => !isFileAccepted(file, accept))) {
+      setErrorMessage(`You provided an unsupported file type. Supported file types are: ${accept}`);
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (uploadType === 'single' && inputFiles.length > 1) {
+      setErrorMessage('You can upload only one file.');
```

Comment:
Check these error messages with design also once if they are not coming from design already